pkg/cmd/kafka/consumergroup/list/list.go

```diff
@@ -162,19 +159,13 @@ func runList(opts *options) (err error) {
 	}
 
 	switch opts.output {
-	case dump.JSONFormat:
-		data, _ := json.Marshal(consumerGroupData)
-		_ = dump.JSON(opts.IO.Out, data)
-	case dump.YAMLFormat, dump.YMLFormat:
-		data, _ := yaml.Marshal(consumerGroupData)
-		_ = dump.YAML(opts.IO.Out, data)
+	case dump.JSONFormat, dump.YAMLFormat, dump.YMLFormat:
+		dump.PrintDataInFormat(opts.output, consumerGroupData, opts.IO.Out)
 	default:
```

Comment:
This is a very convoluted way of simple
```
if opts.output == "table" {
  rows := mapConsumerGroupResultsToTableFormat(consumerGroups)}
  dump.Table(opts.IO.Out, rows)
} else {
   dump.PrintDataInFormat(opts.output, consumerGroupData, opts.IO.Out)
}
```

Reply:
Actually in Go, if/else statements are uncommon, and generally switch statements are recommended over if/else: https://golang.org/doc/effective_go#switch

We can still use a switch, but swap the order to mimic the if/else example by @wtrocki

Reply:
Done